src/main/java/org/gridsuite/voltageinit/server/service/VoltageInitResultContext.java

```diff
@@ -40,36 +43,51 @@ private static String getNonNullHeader(MessageHeaders headers, String name) {
+        Map<String, Double> voltageLevelsIdsRestricted = new HashMap<>();
+        if (headers.containsKey(VOLTAGE_LEVELS_IDS_RESTRICTED) && headers.get(VOLTAGE_LEVELS_IDS_RESTRICTED) != null) {
+            try {
+                TypeReference<Map<String, Double>> typeRef = new TypeReference<>() { };
+                voltageLevelsIdsRestricted = objectMapper.readValue((String) headers.get(VOLTAGE_LEVELS_IDS_RESTRICTED), typeRef);
+            } catch (JsonProcessingException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
+        UUID parametersUuid = null;
+        if (headers.containsKey(PARAMETERS_UUID_HEADER)) {
+            parametersUuid = UUID.fromString(getNonNullHeader(headers, PARAMETERS_UUID_HEADER));
+        }
+        UUID reportUuid = null;
+        if (headers.containsKey(REPORT_UUID_HEADER)) {
+            reportUuid = UUID.fromString(getNonNullHeader(headers, REPORT_UUID_HEADER));
+        }
         String reporterId = headers.containsKey(REPORTER_ID_HEADER) ? (String) headers.get(REPORTER_ID_HEADER) : null;
         String reportType = headers.containsKey(REPORT_TYPE_HEADER) ? (String) headers.get(REPORT_TYPE_HEADER) : null;
         VoltageInitRunContext runContext = new VoltageInitRunContext(networkUuid, variantId, receiver, reportUuid, reporterId, reportType, userId, parametersUuid, voltageLevelsIdsRestricted);
         return new VoltageInitResultContext(resultUuid, runContext);
     }
 
-    public Message<String> toMessage() {
-        return MessageBuilder.withPayload("")
-                .setHeader("resultUuid", resultUuid.toString())
-                .setHeader("networkUuid", runContext.getNetworkUuid().toString())
-                .setHeader(PARAMETERS_UUID_HEADER, runContext.getParametersUuid() != null ? runContext.getParametersUuid().toString() : null)
-                .setHeader(VARIANT_ID_HEADER, runContext.getVariantId())
-                .setHeader(HEADER_RECEIVER, runContext.getReceiver())
-                .setHeader(HEADER_USER_ID, runContext.getUserId())
-                .setHeader(REPORT_UUID_HEADER, runContext.getReportInfos().reportUuid() != null ? runContext.getReportInfos().reportUuid().toString() : null)
-                .setHeader(REPORTER_ID_HEADER, runContext.getReportInfos().reporterId())
-                .setHeader(REPORT_TYPE_HEADER, runContext.getReportInfos().computationType())
-                .setHeader(VOLTAGE_LEVELS_IDS_RESTRICTED, runContext.getVoltageLevelsIdsRestricted())
-                .build();
+    @Override
+    protected Map<String, String> getSpecificMsgHeaders(ObjectMapper objectMapper) {
+        Map<String, String> specificMsgHeaders = new HashMap<>();
+        if (getRunContext().getParametersUuid() != null) {
+            specificMsgHeaders.put(PARAMETERS_UUID_HEADER, getRunContext().getParametersUuid().toString());
+        }
+        if (getRunContext().getVoltageLevelsIdsRestricted() != null && objectMapper != null) {
```

Comment:
```suggestion
        if (getRunContext().getVoltageLevelsIdsRestricted() != null) {
```
ObjectMapper is a bean so it should be always available, we should ignore check null to detect an error in which the bean is not provided. 
Imaging the voltage level ids are provided while we ignore the conversion due to the lack of a bean, that leads to a false positive for the run

Reply:
Agreed; I still had in mind the fact that ObjectMapper was null in worker service before the previous commit.

But those times are over : Done; test removed.